ibis/backends/oracle/__init__.py

```diff
@@ -269,37 +455,42 @@ def _metadata(self, query: str) -> Iterable[tuple[str, dt.DataType]]:
                 typ = dt.Decimal(precision=precision, scale=scale, nullable=nullable)
 
             else:
-                typ = OracleType.from_string(type_string, nullable=nullable)
+                typ = self.compiler.type_mapper.from_string(
+                    type_string, nullable=nullable
+                )
             yield name, typ
 
-    def _table_from_schema(
-        self,
-        name: str,
-        schema: sch.Schema,
-        temp: bool = False,
-        database: str | None = None,
-        **kwargs: Any,
-    ) -> sa.Table:
-        if temp:
-            kwargs["oracle_on_commit"] = "PRESERVE ROWS"
-        t = super()._table_from_schema(name, schema, temp, database, **kwargs)
-        if temp:
-            atexit.register(self._clean_up_tmp_table, t)
-        return t
+    def _fetch_from_cursor(self, cursor, schema: sch.Schema) -> pd.DataFrame:
+        import pandas as pd
+
+        from ibis.backends.oracle.converter import OraclePandasData
+
+        try:
+            df = pd.DataFrame.from_records(
+                cursor, columns=schema.names, coerce_float=True
+            )
+        except Exception:
+            # clean up the cursor if we fail to create the DataFrame
+            #
+            # in the sqlite case failing to close the cursor results in
+            # artificially locked tables
+            cursor.close()
+            raise
+        df = OraclePandasData.convert_table(df, schema)
```

Comment:
We have this method in many of the backends. Nothing to do here except note for a follow up to perhaps generalize a bit.